host/src/server/api/v3/mod.rs

```diff
@@ -0,0 +1,166 @@
+use axum::{response::IntoResponse, Json, Router};
+use raiko_lib::prover::Proof;
+use raiko_tasks::TaskStatus;
+use serde::{Deserialize, Serialize};
+use utoipa::{OpenApi, ToSchema};
+use utoipa_scalar::{Scalar, Servable};
+use utoipa_swagger_ui::SwaggerUi;
+
+use crate::{
+    server::api::v1::{self, GuestOutputDoc},
+    ProverState,
+};
+
+mod proof;
+
+#[derive(OpenApi)]
+#[openapi(
+    info(
+        title = "Raiko Proverd Server API",
+        version = "3.0",
+        description = "Raiko Proverd Server API",
+        contact(
+            name = "API Support",
+            url = "https://community.taiko.xyz",
+            email = "info@taiko.xyz",
+        ),
+        license(
+            name = "MIT",
+            url = "https://github.com/taikoxyz/raiko/blob/main/LICENSE"
+        ),
+    ),
+    components(
+        schemas(
+            raiko_core::interfaces::ProofRequestOpt,
+            raiko_core::interfaces::ProverSpecificOpts,
+            crate::interfaces::HostError,
+            GuestOutputDoc,
+            ProofResponse,
+            TaskStatus,
+            CancelStatus,
+            PruneStatus,
+            Proof,
+            Status,
+        )
+    ),
+    tags(
+        (name = "Proving", description = "Routes that handle proving requests"),
+        (name = "Health", description = "Routes that report the server health status"),
+        (name = "Metrics", description = "Routes that give detailed insight into the server")
+    )
+)]
+/// The root API struct which is generated from the `OpenApi` derive macro.
+pub struct Docs;
+
+#[derive(Debug, Deserialize, Serialize, ToSchema)]
+#[serde(untagged)]
+pub enum ProofResponse {
+    Status {
+        /// The status of the submitted task.
+        status: TaskStatus,
+    },
+    Proof {
+        /// The proof.
+        proof: Proof,
+    },
+}
+
+#[derive(Debug, Deserialize, Serialize, ToSchema)]
+#[serde(tag = "status", rename_all = "lowercase")]
+pub enum Status {
+    Ok { data: ProofResponse },
+    Error { error: String, message: String },
+}
+
+impl From<Vec<u8>> for Status {
+    fn from(proof: Vec<u8>) -> Self {
+        Self::Ok {
+            data: ProofResponse::Proof {
+                proof: serde_json::from_slice(&proof).unwrap_or_default(),
+            },
+        }
+    }
+}
+
+impl From<Proof> for Status {
+    fn from(proof: Proof) -> Self {
+        Self::Ok {
+            data: ProofResponse::Proof { proof },
+        }
+    }
+}
+
+impl From<TaskStatus> for Status {
+    fn from(status: TaskStatus) -> Self {
+        Self::Ok {
+            data: ProofResponse::Status { status },
```

Comment:
As #378 mentioned, we need to refine this part later.

Reply:
I agree. I'll address this after aggregation

Reply:
BTW: could you please review https://github.com/taikoxyz/raiko/pull/378, let's have error report back first, even with unspecific reason tag.